File: julia/words/test/NumericalTest.jl
#!/usr/bin/julia

module NumericalSuite

using Test
include("../src/Numerical.jl")
using .Numerical

@testset "Numerical Suite" begin
  @testset "Fibonacci" begin
    @test fibonacci(-3) == -1
    @test fibonacci(10) == 55
    @test fibonacci(15) == 610
  end

  @testset "Factorial" begin
    @test factorialy(-3) == -1
    @test factorialy(5) == 120
    @test factorialy(10) == 3628800
  end

  @testset "Greatest Common Divisor" begin
    @test gcdy(48, 18) == 6
    @test gcdy(100, 25) == 25
    @test gcdy(17, 13) == 1
  end
  
  @testset "Least Common Multiple" begin
    @test lcmy(4, 6) == 12
    @test lcmy(21, 6) == 42
    @test lcmy(7, 3) == 21
  end
end

end

File: dart/words/test/numerical_test.dart
import 'package:words/numerical.dart';
import 'package:test/test.dart';

void main() {

  group('Numerical', () {
    test('Fibonacci function', () {
      expect(fibonacci(-3), equals(-1));
      expect(fibonacci(10), equals(55));
      expect(fibonacci(15), equals(610));
    });

    test('Factorial function', () {
      expect(factorial(-1), equals(-1));
      expect(factorial(5), equals(120));
      expect(factorial(10), equals(3628800));
    });

    test('Greatest Common Divisor function', () {
      expect(gcd(48, 18), equals(6));
      expect(gcd(100, 25), equals(25));
      expect(gcd(17, 13), equals(1));
    });

    test('Lowest Common Multiple', () {
      expect(lcm(4, 6), equals(12));
      expect(lcm(21, 6), equals(42));
      expect(lcm(7, 3), equals(21));
    });
  });
}